lib/elastic_apm/normalizers/rails/active_record.rb

```diff
@@ -57,6 +57,18 @@ def normalize(_transaction, _name, payload)
         private
 
         def subtype_for(payload)
+          if payload[:connection_id]
+            payload[:connection] ||= begin
```

Comment:
I went back and forth on this idea. It seems like it actually is our own object, we subscribed to notifications from ActiveRecord and they sent us a copy. If I receive a magazine in the mail every month because I asked for it, it's my magazine. We read it and throw it away, don't we?

Maybe my analogy doesn't fit, but it doesn't make any difference to me in the end. I think I went this route because it required less code overall. I'll refactor this method to skip the mutation. I also assume, based on your suggestion, that you think we should remove the `::ActiveRecord::Base.connection_config[:adapter]` check? I mentioned that it was pretty much redundant, as I don't expect there to be a case where that code is reached anyways.